pkg/csi/service/wcp/controller.go

```diff
@@ -527,9 +531,22 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 					TopologyRequirement: topologyRequirement,
 					Vc:                  vc})
 			if err != nil {
+				// Delete the volume created before returning with an error to avoid leaving behind orphan volumes.
+				_, deleteVolumeError := common.DeleteVolumeUtil(ctx, c.manager.VolumeManager,
```

Comment:
I think we are persisting volume in the cache. so next time the duplicate request comes for the same volume, we do not make another call to VC. deleting volume here might affect that workflow. Can you check?

Can we just delete the volume only when we find volume is created on the datastore accessible to multiple zones, and leave it for other cases, so in the next call we can directly jump to finding volume's accessbility

Reply:
Changed the code to delete volume only when retrying won't help.